File: src/main/scala/br/com/fabioaraujo/calcula/produto/service/ItemService.scala
package br.com.fabioaraujo.calcula.produto.service

import br.com.fabioaraujo.calcula.produto.dao.ItemRepository
import br.com.fabioaraujo.calcula.produto.dto.Item
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class ItemService(@Autowired val itemRepository: ItemRepository) {
  def listItens(): java.lang.Iterable[Item] = {
    itemRepository.findAll
  }

  def getItem(id: Long): Item = {
    itemRepository.findOne(id)
  }

  def createItem(item: Item): Long = {
    itemRepository.save(item)
    item.id
  }
}

File: src/main/java/br/com/fabioaraujo/calcula/produto/service/ItemService.java
package br.com.fabioaraujo.calcula.produto.service;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import br.com.fabioaraujo.calcula.produto.dao.ItemRepository;
import br.com.fabioaraujo.calcula.produto.dto.Item;

@Service
public class ItemService {
	@Autowired 
	private  ItemRepository itemRepository;

	public Iterable<Item> listItens(){
		return itemRepository.findAll();
	}

	public Item getItem(Long id) {

		return itemRepository.findById(id).orElse(null);
	}

	public Long createItem(Item item) {
		itemRepository.save(item);
		return item.getId();
	}
}